### PR TITLE
Feat/support searchbox focus - React-SearchBox

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**Description** :
- Support focusShortcuts prop to focus input field using custom keyboard shortcuts.
- support autoFocus prop for the input search field.
- Support addonBefore & addonAfter prop to render custom jsx before & after the input search field.
- Support expandSuggestionsContainer prop to expand/ limit the suggestions dropdown container width to whole input wrapper/ input field.

[Notion Link](https://www.notion.so/appbase/Support-searchbox-focus-with-keyboard-shortcut-3010b391595d4d0495a65e99f9591f9a)

[Docs PR Link](https://github.com/appbaseio/Docs/pull/193)

[Loom Demo: focusShortcuts, autoFocus, addonBefore, addonAfter - props](https://www.loom.com/share/3cf08ca3ba0b47e992a0cf1acf628ae4)

[Loom Demo: expandSuggestionsContainer - prop](https://www.loom.com/share/f196fa3307fc4972bc2238d171290e15)

### CODE REVIEW

File: packages/react-searchbox/src/components/SearchBox.js [hunk 1]
```diff
@@ -63,20 +70,45 @@ class SearchBox extends React.Component {
         stateChanges: true
       });
     }
+
+    // dynamically import hotkey-js
+    if (!isEmpty(focusShortcuts)) {
+      this.hotKeyCombinationsUsed = false;
+      for (let index = 0; index < focusShortcuts.length; index += 1) {
+        if (
```
Comment (mohdashraf010897) [inline packages/react-searchbox/src/components/SearchBox.js]: use the util and optimize the code for importing hotkeys-js

File: packages/react-searchbox/src/components/SearchBox.js [hunk 2]
```diff
@@ -63,20 +70,45 @@ class SearchBox extends React.Component {
...
+          focusShortcuts[index].indexOf('+') !== -1
+        ) {
+          this.hotKeyCombinationsUsed = true;
+          break;
+        }
+      }
+      if (this.hotKeyCombinationsUsed) {
+        import('hotkeys-js').then(module => {
+          this.hotkeys = module.default;
+        });
```
Comment (mohdashraf010897) [inline packages/react-searchbox/src/components/SearchBox.js]: handle errors using catch

File: packages/react-searchbox/src/components/SearchBox.js [hunk 3]
```diff
@@ -439,6 +489,196 @@ class SearchBox extends React.Component {
...
+    // for hotkeys' combinations such as 'cmd+k', 'ctrl+shft+a', etc, we use hotkeys-js
+    if (this.hotKeyCombinationsUsed) {
+      this.hotkeys(
+        parseFocusShortcuts(this.props.focusShortcuts).join(','),
+        (event, handler) => {
+          // Prevent the default refresh event under WINDOWS system
+          event.preventDefault();
+          const elt = e.target || e.srcElement;
+          const tagName = elt.tagName;
+          if (elt.isContentEditable || tagName === 'INPUT') {
```
Comment (mohdashraf010897) [inline packages/react-searchbox/src/components/SearchBox.js]: test with another textarea/ input field and check if focus still works

File: packages/react-searchbox/src/components/SearchBox.js [hunk 4]
```diff
@@ -439,6 +489,196 @@ class SearchBox extends React.Component {
...
+            return;
+          }
+          this.searchInputField.current.focus();
+        }
+      );
+      return;
+    }
+    const shortcuts = this.props.focusShortcuts.map(key => {
+      if (typeof key === 'string') {
+        return isNumeric(key) ? +key : key.toUpperCase().charCodeAt(0);
```
Comment (mohdashraf010897) [inline packages/react-searchbox/src/components/SearchBox.js]: use parseInt insted of unary operator

File: packages/react-searchbox/src/components/SearchBox.js [hunk 5]
```diff
@@ -439,6 +489,196 @@ class SearchBox extends React.Component {
...
+      }
+      return key;
+    });
+    const elt = e.target || e.srcElement;
+    const tagName = elt.tagName;
+    if (elt.isContentEditable || tagName === 'INPUT') {
+      // already in an input
+      return;
+    }
+    let which = e.which || e.keyCode;
```
Comment (mohdashraf010897) [inline packages/react-searchbox/src/components/SearchBox.js]: https://stackoverflow.com/questions/29811465/why-does-the-slash-key-not-have-an-ascii-value-of-47/29811987 
give ref to make it user understandable

File: packages/react-searchbox/src/utils/helper.js [hunk 6]
```diff
@@ -88,17 +88,72 @@ export function isEqual(x, y) {
...
+  let chrCode = which - 48 * Math.floor(which / 48);
+  return String.fromCharCode(which >= 96 ? chrCode : which);
+}
+
+// used for parsing focusshortcuts for keycodes passed as string, eg: 'ctrl+/' is same as 'ctrl+47'
+// returns focusShortcuts containing appropriate key charsas depicted on keyboards
+export function parseFocusShortcuts(focusShortcutsArray) {
+  if (isEmpty(focusShortcutsArray)) return [];
+
+  let parsedFocusShortcutsArray = [];
```
Comment (mohdashraf010897) [inline packages/react-searchbox/src/utils/helper.js:140]: add comment to convey the context

**Review by bietkul — APPROVED** [review]
lgtm 👍

**Pull request merged**